src/Components/OATHeader/OATHeader.tsx

```diff
@@ -1,372 +1,257 @@
-            if (files.length > 0) {
-                const filesErrors = [];
-                let modelsMetadataReference = null;
-                for (const current of files) {
-                    const content = await current.text();
-                    const validJson = safeJsonParse(content);
-                    if (validJson) {
-                        newModels.push(validJson);
-                    } else {
-                        filesErrors.push(
-                            t('OATHeader.errorFileInvalidJSON', {
-                                fileName: current.name
-                            })
+            const handleFileListChanged = async (files: Array<File>) => {
+                logDebugConsole(
+                    'debug',
+                    '[IMPORT] [START] Parsing files. {files}',
+                    files
+                );
+                const newModels: DtdlInterface[] = [];
+                if (files.length > 0) {
+                    const filesErrors = [];
+                    let modelsMetadataReference = null;
+                    for (const current of files) {
+                        const content = await current.text();
+                        const validJson = safeJsonParse<DtdlInterface | null>(
+                            content
                         );
-                        break;
+                        if (validJson) {
+                            newModels.push(validJson);
+                        } else {
+                            filesErrors.push(
+                                t('OATHeader.errorFileInvalidJSON', {
+                                    fileName: current.name
+                                })
+                            );
+                            break;
+                        }
                     }
-                }
 
-                const combinedModels = [...models, ...newModels];
-                const error = await parseModels(combinedModels);
+                    const combinedModels = [
+                        ...oatPageState.currentOntologyModels,
+                        ...newModels
+                    ];
+                    const error = await parseModels(combinedModels);
 
-                const modelsCopy = deepCopy(models);
-                for (const model of newModels) {
-                    // Check if model already exists
-                    const modelExists = modelsCopy.find(
-                        (m) => m['@id'] === model['@id']
+                    const modelsCopy = deepCopy(
+                        oatPageState.currentOntologyModels
                     );
-                    if (!modelExists) {
-                        modelsCopy.push(model);
+                    // remove duplicates
+                    for (const model of newModels) {
+                        // Check if model already exists
+                        const modelExists = modelsCopy.find(
+                            (m) => m['@id'] === model['@id']
```

Comment:
Could we move this check to the `if (validJson) { newModels.push...` section to remove an iteration of newModels/files array?

Reply:
WOrking on it.

Reply:
Done.